src/greynirseq/nicenlp/tasks/document_translation_from_pretrained_bart.py

```diff
@@ -0,0 +1,385 @@
+            " from the JSONL language to the language string expected by the model, e.g. 'is:is_IS,en:en_XX'"
+        },
+    )
+
+
+@register_task("document_translation_from_pretrained_bart", dataclass=DocumentTranslationFromPretrainedBARTConfig)
+class DocumentTranslationFromPretrainedBART(TranslationFromPretrainedBARTTask):
+    """Task for training multi sentence translation models from pre-trained BART models."""
+
+    def __init__(
+        self, cfg: DocumentTranslationFromPretrainedBARTConfig, the_dict: Dictionary, language_mappings: Dict[str, str]
+    ):
+        super().__init__(cfg, src_dict=the_dict, tgt_dict=copy.deepcopy(the_dict))
+        # this is for typing only
+        self.the_dict = the_dict
+        self.language_mappings = language_mappings
+
+    @classmethod
+    def setup_task(cls, cfg: DocumentTranslationFromPretrainedBARTConfig, **kwargs):
+        """Setup the task (e.g., load dictionaries).
+
+        Args:
+            args (argparse.Namespace): parsed command-line arguments
+        """
+        if cfg.dict_path == "":
+            raise ValueError("Must specify a dictionary path")
+        if cfg.data_language_mappings == "":
+            raise ValueError("Must specify languages to train on")
+        the_dict = cls.load_dictionary(cfg.dict_path)
+        logger.info("dictionary: {} types".format(len(the_dict)))
+        # langcode and translation direction, e.g. "en:en_XX,is:is_IS"
+        language_mappings_pairs = cfg.data_language_mappings.split(",")
+        language_mappings = {}
+        for pair in language_mappings_pairs:
+            parts = pair.split(":")
+            if len(parts) != 2:
+                raise ValueError("Invalid language mapping: {}".format(pair))
+            language_mappings[parts[0]] = parts[1]
+
+        return cls(cfg, the_dict, language_mappings)
+
+    def load_dataset(self, split: str, epoch=1, combine=False, **kwargs):
+        """Load a given dataset split.
+
+        This function is called once per train split (e.g., "train") and multiple times per validation split.
+        This function has deviated a lot from the original fairseq implementation.
+
+        Args:
+            split (str): The value of --train-subset OR
+                --valid-subset which has been split on "," OR
+                --test-subset which has been split on "," CLI args.
+                Each is a comma-separated list of dataset names.
+                This method is called separately for each subset.
+        """
+        self.cfg = cast(DocumentTranslationFromPretrainedBARTConfig, self.cfg)
+        # this is for sharding
+        paths = utils.split_paths(self.cfg.data)  # type: ignore
+        assert len(paths) > 0
+        data_dir_path = paths[(epoch - 1) % len(paths)]
+
+        # if a split contains a comma, we should crash - since that is no longer supported
+        assert "," not in split, "Split should not contain a comma"
```

Comment:
But when split contains the cli input of --train-subset, it is allowed to have commas according to the function documentation